src/CBT.Core/Tasks/RestoreModules.cs

```diff
@@ -357,6 +361,24 @@ private bool TryRestorePackages(out ModuleRestoreInfo moduleRestoreInfo)
 
             string moduleRestoreInfoFile = Path.Combine(Path.GetTempPath(), $"{Guid.NewGuid():N}.tmp");
 
+            try
+            {
+                FileVersionInfo fileVersionInfo = FileVersionInfo.GetVersionInfo(RestoreCommand);
+
+                if (fileVersionInfo.ProductMajorPart >= 4 && RestoreCommandArguments.IndexOf("-MSBuildPath", StringComparison.OrdinalIgnoreCase) == -1)
+                {
+                    string msbuildPathArgument = $" -MSBuildPath \"{MSBuildBinPath}\"";
```

Comment:
I don't see where MSBuildBinPath is guaranteed to be set to a value and can't just be an empty string being passed in.  So perhaps do a string isnullorwhitspace check as well.

Reply:
I think `$(MSBuildBinPath)` is always set by MSBuild right?